lib/ansible/module_utils/aws/core.py

```diff
@@ -169,12 +169,29 @@ def fail_json_aws(self, exception, msg=None):
         except AttributeError:
             response = None
 
+        boto3_version, botocore_version = self._gather_versions()
```

Comment:
There's a lot of repetition here.

If _gather_versions returned `{}` when `HAS_BOTO3` is false, and `{'boto3_version': boto3.__version__, 'botocore_version': botocore.__version__}` otherwise, we could just do:

```
failure = dict(msg=message, exception=last_traceback)
failure.update(self._gather_versions)
if response is not None:
    failure.update(**camel_dict_to_snake_dict(response)
self._module.fail_json(**failure)
```

Reply:
Yeah, that does seem nicer :grinning: